src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c

```diff
@@ -23,26 +21,133 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
 };
 */
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40080.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B4009C.asm")
+void func_80B40100(BgGoronOyu* _this, GlobalContext* globalCtx);
+void func_80B400C8(BgGoronOyu* _this, GlobalContext* globalCtx);
+void func_80B401F8(BgGoronOyu* _this, GlobalContext* globalCtx);
+void func_80B40308(BgGoronOyu* _this, GlobalContext* globalCtx);
+void func_80B40394(BgGoronOyu* _this, GlobalContext* globalCtx);
+void func_80B40160(BgGoronOyu* _this, GlobalContext* globalCtx);
+
+void func_80B40080(BgGoronOyu* _this) {
+    _this->unk17E = 1;
+    _this->actionFunc = &func_80B400C8;
+}
+
+void func_80B4009C(BgGoronOyu* _this) {
+    BgGoronOyuActionFunc actionCallback;
+
+    _this->unk17E = 0;
+    _this->initialActorCutscene = _this->polyActor.actor.cutscene;
+
+    actionCallback = &func_80B40100;
+    _this->actionFunc = actionCallback;
+    _this->flt164 = 20.0f;
+}
+
+void func_80B400C8(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    func_80B40308(_this, globalCtx);
+    func_80B401F8(_this, globalCtx);
+}
+
+void func_80B40100(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    s16 actorCutscene;
+    s16 actorCutscene_2;
+
+    actorCutscene = _this->initialActorCutscene;
+
+    if (ActorCutscene_GetCanPlayNext(actorCutscene) != 0) {
+        actorCutscene_2 = _this->initialActorCutscene;
+        ActorCutscene_StartAndSetUnkLinkFields(actorCutscene_2, &_this->polyActor.actor);
```

Comment:
```suggestion
        ActorCutscene_StartAndSetUnkLinkFields(this->initialActorCutscene, &this->polyActor.actor);
```
Is this temp needed?